mettle/src/stdapi/audio/linux_output.c

```diff
@@ -0,0 +1,30 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/wait.h>
+
+#include <mettle.h>
+#include "channel.h"
+
+#include "output.h"
+
+int terminate_audio_file(struct channel *c)
+{
+    // File was uploaded, play it and free the buffer
+    context *ctx = channel_get_ctx(c);
+
+    // Open pipe, we will send the content to the aplay command
+    // Similar to `cat test.wav | aplay`
+    FILE *pipe_fp = popen("aplay -q", "w");
+    if (pipe_fp == NULL) {
+        return -1;
+    }
+
+    if (fwrite(ctx->buffer, 1, ctx->size, pipe_fp) != ctx->size) {
+        return -1;
+    }
+
+    free(ctx->buffer);
+    free(ctx);
+
+    return WEXITSTATUS(pclose(pipe_fp));
```

Comment:
This should do the trick, however I ain't sure how to do this in the apple part.

Reply:
Actually this might not even work since `Channel._close` would always return `true`.